### PR TITLE
DATAUP-246, css to scss: scss transformation!

### PR DESCRIPTION
**Pull request by ialarmedalien**
# Description of PR purpose/changes

* This PR converts the existing css files to scss, and includes some minor refactoring. Further refactoring can occur once this change has been tested and deployed.

What I did:
- installed [compass](http://compass-style.org), which includes a sass converter that can take in css and spit out scss. Using that, I generated the files in `kbase-extension/scss/partials` from our existing css.
- moved all font declarations to a single file
- created variables for the typeface combos that we use and substituted them throughout
- updated the gruntfile to remove all the code for concatenating files, etc., and added in new tasks and npm commands for compiling css.

The work is split up into logical commits, so if you want to view things commit by commit to relive the exciting experiences that I had, you can do so!

# Jira Ticket / Issue #
e.g. https://kbase-jira.atlassian.net/browse/DATAUP-246

# Testing Instructions
* Details for how to test the PR: 
- [ ] Tests pass in travis and locally 
- [ ] Changes available by spinning up a local narrative and clicking around. Please ensure that the instructions in the docs included in the PR work in your environment so I can amend them if not.

# Dev Checklist:

- [ ] My code follows the guidelines at https://sites.google.com/truss.works/kbasetruss/development
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: Gruntfile.js [hunk 1]
```diff
@@ -227,83 +193,61 @@ module.exports = function(grunt) {
...
         // check the source css files are not completely horrendous
+        // config to be updated for scss files
         stylelint: {
             options: stylelintOptions,
-            kbaseCss: {
-                src: staticFilesNoConcat,
-            },
+            concat: {
+                src: kbaseStaticDir + 'css/*.css',
+            }
```
Comment (ialarmedalien) [inline Gruntfile.js:203]: This will be updated in a future PR as the stylelint options that were in place were for css files.

File: Gruntfile.js [hunk 2]
```diff
@@ -227,83 +193,61 @@ module.exports = function(grunt) {
...
+                                    "exclude": true
+                                },
+                            }
+                        ]),
                     ],
                 },
                 src: [
                     'kbase-extension/static/kbase/css/*_concat.css',
+                    'kbase-extension/static/kbase/css/appCell.css',
+                    'kbase-extension/static/kbase/css/editorCell.css'
```
Comment (ialarmedalien) [inline Gruntfile.js:229]: these files are now produced by scss compilation. Ideally they'd all be merged into one big file... possibly in a future PR.

File: Gruntfile.js [hunk 3]
```diff
@@ -227,83 +192,58 @@ module.exports = function(grunt) {
...
-            // a single file containing all the KBase css!
-            allCss: {
-                src: allCssFiles,
-                dest: kbaseStaticDir + 'css/all_concat.css',
-                nonull: true,
-            }
-        },
 
         // check the source css files are not completely horrendous
+        // config to be updated for scss files
```
Comment (ialarmedalien) [inline Gruntfile.js:197]: This task doesn't do anything at the moment as I haven't yet configured the stylelint settings for scss files.

File: package.json [hunk 4]
```diff
@@ -64,9 +67,9 @@
   "scripts": {
     "bower": "bower install",
     "test": "grunt test",
-    "build": "grunt build_css",
-    "stylelint": "stylelint --fix 'kbase-extension/static/kbase/css/*.css' 'kbase-extension/static/kbase/custom/custom.css'",
-    "update_browserslist": "npx browserslist@latest --update-db"
+    "update_browserslist": "npx browserslist@latest --update-db",
+    "compile_css": "sass kbase-extension/scss/:kbase-extension/static/kbase/css/ && grunt postcss:concat",
+    "sassv": "sass --version"
```
Comment (ialarmedalien) [inline package.json:72]: useful check that you've got sass installed

File: package.json [hunk 5]
```diff
@@ -64,9 +67,9 @@
   "scripts": {
     "bower": "bower install",
     "test": "grunt test",
-    "build": "grunt build_css",
-    "stylelint": "stylelint --fix 'kbase-extension/static/kbase/css/*.css' 'kbase-extension/static/kbase/custom/custom.css'",
-    "update_browserslist": "npx browserslist@latest --update-db"
+    "update_browserslist": "npx browserslist@latest --update-db",
+    "compile_css": "sass kbase-extension/scss/:kbase-extension/static/kbase/css/ && grunt postcss:concat",
```
Comment (ialarmedalien) [inline package.json:71]: ain't that beautiful?

**Review by eamahanna — APPROVED** [review]

**Pull request merged**